### PR TITLE
2 developer documentation

### PR DESCRIPTION
**Pull request by fdiblen**
Adds developer documentation.
Refs: #2

### CODE REVIEW

File: README.dev.rst [hunk 1]
```diff
@@ -0,0 +1,195 @@
+``fairtally`` developer documentation
+=====================================
+
+If you're looking for user documentation, go `here <README.rst>`_.
```
Comment (sverhoeven) [inline README.dev.rst:4]: Can you link from README.rst to here

**Review by sverhoeven — CHANGES_REQUESTED** [review]

File: README.dev.rst (hunk 1 above)
Comment (sverhoeven) [reply README.dev.rst:4]: Its there now

File: README.dev.rst [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+Running the tests
+-----------------
+
+Running the tests requires an activated virtualenv with the development tools installed.
+
+.. code:: shell
+
+    # unit tests with mocked representations of repository behavior
+    pytest
+    pytest tests/
```
Comment (sverhoeven) [inline README.dev.rst:49]: Duplicate
```suggestion
```

File: README.dev.rst [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+
```
Comment (sverhoeven) [inline README.dev.rst:50]:
```suggestion
The test coverage report is available in `htmlcov/index.html`.
```

File: README.dev.rst [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+    prospector
+
+    # recursively check import style for the fairtally module only
+    isort --recursive --check-only fairtally
+
+    # recursively check import style for the fairtally module only and show
+    # any proposed changes as a diff
+    isort --recursive --check-only --diff fairtally
+
+    # recursively fix  import style for the fairtally module only
```
Comment (sverhoeven) [inline README.dev.rst:68]:
```suggestion
    # recursively fix import style for the fairtally module only
```

File: README.dev.rst [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+    isort --recursive fairtally
+
+.. code:: shell
+
+    # requires activated virtualenv with development tools
+    prospector && isort --recursive --check-only fairtally
```
Comment (sverhoeven) [inline README.dev.rst:74]: Duplicate or explain what this does different from above.
```suggestion
```

File: README.dev.rst [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+    git config --local core.hooksPath .githooks
+
+Versioning
+----------
+
+Bumping the version across all files is done with bump2version, e.g.
+
+.. code:: shell
+
+    bump2version minor
```
Comment (sverhoeven) [inline README.dev.rst:85]: Does not work for me

```
usage: bumpversion [-h] [--config-file FILE] [--verbose] [--list] [--allow-dirty] [--parse REGEX] [--serialize FORMAT]
                   [--search SEARCH] [--replace REPLACE] [--current-version VERSION] [--no-configured-files] [--dry-run]
                   --new-version VERSION [--commit | --no-commit] [--tag | --no-tag] [--sign-tags | --no-sign-tags]
                   [--tag-name TAG_NAME] [--tag-message TAG_MESSAGE] [--message COMMIT_MSG] [--commit-args COMMIT_ARGS]
                   part [file [file ...]]
bumpversion: error: the following arguments are required: --new-version
```

Comment (sverhoeven) [reply README.dev.rst:85]: I have `bumpversion                   0.6.0`

File: README.dev.rst [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+
+
+Making a release
+----------------
+
+Preparation
+^^^^^^^^^^^
+
+1. Update the ``CHANGELOG.srt``
```
Comment (sverhoeven) [inline README.dev.rst:94]:
```suggestion
1. Update the ``CHANGELOG.rst``
```

File: README.dev.rst [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+2. Verify that the information in ``CITATION.cff`` is correct, and that ``.zenodo.json`` contains equivalent data
+3. Make sure the version has been updated.
+4. Run the unit tests with ``pytest tests/``
```
Comment (sverhoeven) [inline README.dev.rst:97]: Consistent command
```suggestion
4. Run the unit tests with ``pytest``
```

File: README.dev.rst [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+.. code:: shell
+
+    # Back to the first terminal,
+    # FINAL STEP: upload to PyPI (requires credentials)
+    twine upload dist/*
+
+GitHub
+^^^^^^
+
+Don't forget to also make a release on GitHub.
```
Comment (sverhoeven) [inline README.dev.rst:161]:
```suggestion
Don't forget to also `make a release on GitHub <https://github.com/fair-software/fairtally/releases/new>`_.
```

**Review by sverhoeven — CHANGES_REQUESTED** [review]
Most of text look OK to me.

Can figure out why bumpversion is not working?

File: README.dev.rst (hunk 6 above)
Comment (fdiblen) [reply README.dev.rst:85]: We don't have a bump2version setup. I created an issue.

**Review by sverhoeven — APPROVED** [review]
Looks like bumpversion issue will be handled separately in #54

**Pull request merged**